ethereum/PFP.sol

```diff
@@ -133,37 +142,38 @@ contract PFP is ERC721A, Ownable, VRFConsumerBase {
         }
         string memory currentBaseURI = _baseURI();
         return bytes(currentBaseURI).length > 0
-            ? string(abi.encodePacked(currentBaseURI, ((tokenId+randStartPos) % MAX_TOKENS).toString(), baseExtension))
+            ? string(abi.encodePacked(currentBaseURI, ((tokenId+randStartPos) % MAX_TOKENS).toString(), BASE_EXTENSION))
             : "";
     }
 
-    function claimTokens(bytes32[] calldata merkleProof, uint256 numberOfTokens, uint256 allowance ) public returns(string memory) {
+    function claimTokens(bytes32[] calldata merkleProof, uint256 numberOfTokens, uint256 allowance ) external {
         address[] memory coldWallets = EPS.getAddresses(msg.sender, _passAddress, 1, true, true);
-        string memory result;
-        bool isVerified = false;
         require(totalSupply() + numberOfTokens <= MAX_TOKENS, "Claim would exceed max supply of tokens!");
         for (uint256 i = 0; i < coldWallets.length; i++) {
             address coldWallet = coldWallets[i];
-            if (MerkleProof.verify(merkleProof, snapshotMerkleRoot, keccak256(abi.encodePacked(msg.sender, '_', allowance)))) {
+            if (MerkleProof.verify(merkleProof, snapshotMerkleRoot, keccak256(abi.encodePacked(coldWallet, '_', allowance)))) {
                 if(_claimedAmount[coldWallet] + numberOfTokens <= allowance ) {
                     _safeMint(msg.sender, numberOfTokens);
-                    result = string(abi.encodePacked("Mint is successed!"));
-                    return result;
+                    _claimedAmount[coldWallet] += numberOfTokens;
+                    emit WebaversePFPMint(numberOfTokens);
+                    // Function exit 1: success
+                    return;
                 } else {
-                    isVerified = true;
+                    // Function exit 2: claim exceeds allowance
+                    revert ClaimExceedsAllowance(uint128(numberOfTokens), uint128(allowance - _claimedAmount[coldWallet]));
```

Comment:
Rather than have a successful call and pass back a string I think this should revert. The biggest advantage here is that gas estimators will read this, and warn the user prior to submission that it will fail. If we return the string it's quite confusing for the user, as they have paid the gas and the call has completed but they get no NFTs.